### PR TITLE
Segment analytics

### PR DESCRIPTION
**Pull request by ErinHales**
Closes celo-org/celo-monorepo#6034

## Motivation

Track user activity on Blockscout.
https://docs.google.com/spreadsheets/d/1HYdr9L2JCDgHvuCrpzUce-LMTxzclltG5LO638qTuUQ/edit#gid=447598208

## Other Changes

Adds Cookies Banner
Screen Shot 2021-02-01 at 5 49 46 PM

### CODE REVIEW

File: apps/block_scout_web/lib/block_scout_web/templates/layout/app.html.eex [hunk 1]
```diff
@@ -35,14 +35,17 @@
...
       function applyDarkMode() {
         if (localStorage.getItem("current-color-mode") === "dark") {
           document.body.className += " " + "dark-theme-applied";
           document.body.style.backgroundColor = "#1c1d31";
         }
       }
       window.onload = applyDarkMode()
+
+      // TODO: import this function
+      // initAnalytics('<%= segment_key() %>')
```
Comment (ErinHales) [inline apps/block_scout_web/lib/block_scout_web/templates/layout/app.html.eex:48]: It won't let me import this function, says it cannot find the file 🤔

File: apps/block_scout_web/assets/js/lib/analytics.js [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+function trackEvents () {
+  $(function () {
+    // Page navigation
+    window.addEventListener('locationchange', function () {
+      analytics.page()
+    })
+
+    // Search box click
+    $('[data-selector="search-bar"]').on('click', function () {
+      analytics.track('Search bar clicked')
```
Comment (carterqw2) [inline apps/block_scout_web/assets/js/lib/analytics.js:41]: Is it a legit event name? Could you double check with JB if it's not going to cause any issues, perhaps we should use target names from his document.

File: apps/block_scout_web/assets/js/lib/analytics.js [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+    case path.includes('/tx') && path.includes('/internal_transactions'):
+      return 'transactionInternalTransaction'
+    case path.includes('/tx') && path.includes('/logs'):
+      return 'transactionLogs'
+    case path.includes('/tx') && path.includes('/raw_trace'):
+      return 'transactionRawTrace'
+    case path.includes('/tx') && path.includes('/token_transfers'):
+      return 'transactionTokenTransfers'
+    default:
+      return 'unknown'
```
Comment (aaronmgdr) [inline apps/block_scout_web/assets/js/lib/analytics.js:83]: why not have default be the original path? that way we can see what we missed

File: apps/block_scout_web/assets/js/lib/analytics.js [hunk 4]
```diff
@@ -0,0 +1,257 @@
...
+  }
+}
+
+// returns referrer path, strips domain name from referrer
+function getReferrerPath () {
+  const referrer = document.referrer
+  return referrer.replace('https://explorer.celo.org', '').replace('http://localhost:4000', '')
```
Comment (aaronmgdr) [inline apps/block_scout_web/assets/js/lib/analytics.js:90]: why strip domain?

**Comment by aaronmgdr** [comment]
one question i have is that i see you set up manual page and event tracking, often these analytics packages will do that automatically so want to make sure we are going to be double counting anything.

File: apps/block_scout_web/assets/js/lib/analytics.js (hunk 3 above)
Comment (ErinHales) [reply apps/block_scout_web/assets/js/lib/analytics.js:83]: Ooh, good idea

**Comment by ErinHales** [comment]
Excellent question.  JB has a very specific set of requirements for how he wants events displayed and what data he wants attached to events.  That's why I decided to make a manual "navigation" event instead of using the `analytics.page` function.  If that proves to be an issue, we'll be beefing this up and adding server-side analytics at a future date and can fine-tune the existing system then.

File: apps/block_scout_web/assets/js/lib/analytics.js (hunk 4 above)
Comment (ErinHales) [reply apps/block_scout_web/assets/js/lib/analytics.js:90]: For use inside the `getPageName` function. JB wanted page names displayed in a more readable format.

**Comment by aaronmgdr** [comment]
> Excellent question. JB has a very specific set of requirements for how he wants events displayed and what data he wants attached to events. That's why I decided to make a manual "navigation" event instead of using the `analytics.page` function. If that proves to be an issue, we'll be beefing this up and adding server-side analytics at a future date and can fine-tune the existing system then.

ok cool.  the question i have is not why the custom code but are you certain that there any automatic even tracking is turned off.

**Comment by ErinHales** [comment]
> > Excellent question. JB has a very specific set of requirements for how he wants events displayed and what data he wants attached to events. That's why I decided to make a manual "navigation" event instead of using the `analytics.page` function. If that proves to be an issue, we'll be beefing this up and adding server-side analytics at a future date and can fine-tune the existing system then.
> 
> ok cool. the question i have is not why the custom code but are you certain that there any automatic even tracking is turned off.

Oh, k. Yes, I'm pretty certain that this package doesn't have automatic tracking.

File: apps/block_scout_web/assets/js/lib/analytics.js [hunk 5]
```diff
@@ -0,0 +1,257 @@
...
+function getNetwork () {
+  switch (window.location.host) {
+    case 'explorer.celo.org':
+      return 'mainnet'
+    case 'alfajores-blockscout.celo-testnet.org':
+      return 'alfajores'
+    case 'baklava-blockscout.celo-testnet.org':
+      return 'baklava'
+    default:
+      return 'mainnet'
```
Comment (carterqw2) [inline apps/block_scout_web/assets/js/lib/analytics.js:117]: Maybe it's not a safe default. We have other, instance-specific hostnames, like `https://baklava-blockscout3.celo-testnet.org/` that would fall into this bucket and will be incorrectly attributed to mainnet.

File: apps/block_scout_web/assets/js/lib/analytics.js [hunk 6]
```diff
@@ -130,6 +130,8 @@ function getEntityId () {
 function getNetwork () {
   const host = window.location.host
   switch (true) {
```
Comment (aaronmgdr) [inline apps/block_scout_web/assets/js/lib/analytics.js:132]: interesting i've never seen this done with switch statement setting the switch to true. 

not a value judgement just a oh wow that's possible moment.

Comment (ErinHales) [reply apps/block_scout_web/assets/js/lib/analytics.js:132]: I generally wouldn't do it but there would be SO MANY if/else statements in this file if I didn't

**Comment by aaronmgdr** [comment]
let me know if this is outta scope thoughts on that blue in the consent banner? Im questioning its readability index both with white and dark text. Can just use the standard blockscout background color?

**Comment by ErinHales** [comment]
> let me know if this is outta scope thoughts on that blue in the consent banner? Im questioning its readability index both with white and dark text. Can just use the standard blockscout background color?

Not super important right now. If Taylor gets the new design to me before this is merged, I will update. If not, I think it's more important to get this up and running.

**Review by jcortejoso — APPROVED** [review]
Looks good :+1:

**Pull request merged**